pageserver/src/tenant.rs

```diff
@@ -1236,8 +1240,24 @@ impl Tenant {
         };
 
         for (timeline_id, timeline) in &timelines_to_compact {
+            if tenant_cxt.is_cancelled() {
+                // We were requested to shut down. Stop and return with the progress we
+                // made.
+                break;
+            }
+
+            let timeline_cxt = match timeline.get_context(tenant_cxt) {
```

Comment:
Since we have a hierarchy notion between contexts, I wonder if such straightforward tenant <- timeline relation could automatically merge such things?
If the tenant is cancelled, the timeline context should never allowed to be active.